controllers/snapshot_controller.go

```diff
@@ -152,7 +153,7 @@ func (r *SnapshotReconciler) reconcileDeleteSnapshot(ctx context.Context, obj *d
 
 	if err := r.Cache.DeleteData(ctx, name, obj.Spec.Tag); err != nil {
 		var terr *transport.Error
-		if !errors.As(err, &terr) || !containsManifestNotFoundError(terr.Errors) {
+		if !errors.As(err, &terr) || (terr.StatusCode != http.StatusNotFound && !containsManifestNotFoundError(terr.Errors)) {
```

Comment:
this is very confusing to read 😄 ; could we refactor slightly for readbility?

Reply:
:D :D sure. :)

Reply:
Amended. :))) 👍